### PR TITLE
Refactor flowgraph and related invocation

### LINKED ISSUE #8742: Flowgraph may have goroutine leak when closing
When closing `flowgraph`, some node may running into a corner case that its downstream is closed but it's still trying to send a msg into channel. Then this goroutine can never quit.

See code below:
https://github.com/milvus-io/milvus/blob/0e767f0e1b6d2f644d493c929fbef94f788d6ebd/internal/util/flowgraph/node.go#L114-L124

### PR DESCRIPTION
**Pull request by congqixia**
Fix #8742 
See also #7789

Signed-off-by: Congqi Xia <congqi.xia@zilliz.com>

### CODE REVIEW

File: internal/util/flowgraph/node.go [hunk 1]
```diff
@@ -22,62 +21,58 @@ import (
...
+		case <-nodeCtx.closeCh:
 			return
 		default:
 			// inputs from inputsMessages for Operate
 			inputs := make([]Msg, 0)
 
 			var res []Msg
 			if !nodeCtx.node.IsInputNode() {
-				nodeCtx.collectInputMessages(ctx)
+				nodeCtx.collectInputMessages()
```
Comment (xiaofan-luan) [inline internal/util/flowgraph/node.go:75]: if channel is closed, then inputs could be a empty message list.
However in some of the nodes they check inputs length, that may cause unexpected erros

File: internal/util/flowgraph/node.go [hunk 2]
```diff
@@ -95,35 +90,37 @@ func (nodeCtx *nodeCtx) Start(ctx context.Context, wg *sync.WaitGroup) {
...
 func (nodeCtx *nodeCtx) Close() {
-	// data race with nodeCtx.ReceiveMsg { nodeCtx.inputChannels[inputChanIdx] <- msg }
-	//for _, channel := range nodeCtx.inputChannels {
-	//	close(channel)
-	//	log.Warn("close inputChannel")
-	//}
+	// close Node
 	nodeCtx.node.Close()
+	// notify worker
+	close(nodeCtx.closeCh)
```
Comment (xiaofan-luan) [inline internal/util/flowgraph/node.go:105]: Do we want to close inputChannels? what

**Review by xiaofan-luan — APPROVED** [review]

File: internal/util/flowgraph/node.go (hunk 2 above)
Comment (congqixia) [reply internal/util/flowgraph/node.go:105]: Downstream node closes before current node is a possible situation and was the root cause of goroutine leakage. In `deliverMsg` we have a select to cancel send to channel operation to handle this case.

File: internal/util/flowgraph/node.go (hunk 1 above)
Comment (congqixia) [reply internal/util/flowgraph/node.go:75]: Currently the input msg will never be empty. When closing a node, the corner case might be a list with size of input channels with nil value.

**Pull request merged**